Sdl.Web.Tridion/Mapping/DefaultModelBuilder.cs

```diff
@@ -133,6 +117,56 @@ public virtual void BuildPageModel(ref PageModel pageModel, IPage page, IEnumera
             }
         }
 
+        private RegionModel GetRegionForEntity(RegionModelSet regions, IComponentPresentation cp, RegionModel currentRegion, Dictionary<String,int> duplicateRegionCounter)
+        {
+            MvcData cpRegionMvcData = GetRegionMvcData(cp);
+            string regionName = cpRegionMvcData.RegionName ?? cpRegionMvcData.ViewName;
+            string currentRegionName = currentRegion!=null ? currentRegion.NameWithoutPostfix : null;
+            string duplicateRegionName = null;
+            RegionModel region = null;
+            //If we are still in the same region, reuse it, unless the mvc data is different, 
+            //in which case we start a new region of the same name
+            if (regionName==currentRegionName)
+            {
+                if (currentRegion.MvcData.Equals(cpRegionMvcData))
+                {
+                    return currentRegion;
+                }
+                else
+                {
+                    Log.Debug("Region '{0}' is defined with different MVC data: [{1}] and [{2}]. Starting a new region.", regionName, currentRegion.MvcData, cpRegionMvcData);

```

Comment:
Note that now we can have duplicate regions, a change in MVC data can simply trigger a new region of the same name. This ties in nicely with PR #29 which allows regions to be configurable via CT metadata. In this case we would want to start a new region if the region route values were different